File: scala/b06.scala
import  scala.io.StdIn._

object Main {
    def main(args: Array[String]): Unit= {
        val n = readInt()
        val listA = readLine().split(" ").map(_.toInt)
        val q = readInt()
        val winSum = listA.scanLeft(0){_+_}
        val loseSum = listA.map(a => {
            if (a == 1) 0
            else 1
        }).scanLeft(0){_+_}

        for ( i <- 1 to q) {
            val Array(l,r) = readLine().split(" ").map(_.toInt)
            val wl = (winSum(r)-winSum(l-1)) - (loseSum(r)-loseSum(l-1))
            if (wl > 0) println("win")
            else if (wl < 0) println("lose")
            else println("draw")
        }
    }
}

File: python/b06.py
N = int(input())
A = list(map(int, input().split()))
w = [0]
lo = [0]
for i in range(N):
    if A[i] == 1:
        w.append(w[-1]+1)
        lo.append(lo[-1])
    else:
        w.append(w[-1])
        lo.append(lo[-1]+1)


Q = int(input())
for _ in range(Q):
    l, r = map(int, input().split())
    wl = (w[r] - w[l-1]) - (lo[r] - lo[l-1])
    if wl > 0:
        print("win")
    elif wl < 0:
        print("lose")
    else:
        print("draw")
